### DISCUSSION TITLE
Python tools

### DISCUSSION THREAD

**Discussion by chris48s** [jotaen/xit#53] [Show & tell]
I've been noodling around with some python tooling for [x]it

- https://github.com/chris48s/stage-left is a parser for reading [x]it! checklists into python data structures
- https://github.com/chris48s/xit2md is the first tool I've built which uses that parser. It can convert a [x]it! document into markdown task lists

Very early days, but a functioning parser and one (basic) user-facing tool dogfooding it is a start. I'll continue to update as I work on things more.

**Comment by jotaen** [comment]
🎉  That looks neat. It’s cool that you made the parser a standalone library, which makes it easy to re-use for other Python apps.

I’m happy to link to your project(s) from the [website](https://xit.jotaen.net/), so that other users can find it easily. Just let me know if or when you’d like that.

**Reply by chris48s** [reply]
Thanks. I think I'm going to carry on tinkering and dogfooding for a bit before I ask you to link stuff from there but I'll let you know. Cheers :+1:

**Reply by chris48s** [reply]
Having done a bit more work on these (in particular, writing a proper [docs site](https://chris48s.github.io/stage-left/) for the parser and adding a [CLI tool](https://github.com/chris48s/xit2md#on-the-console) to xit2md), I think it is now worthwhile to add https://github.com/chris48s/stage-left and https://github.com/chris48s/xit2md to the tools section on https://xit.jotaen.net/ if that is ok.

**Reply by jotaen** [reply]
Cool, both projects are linked now!
